trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/extract-query-and-session-data.ts

```diff
@@ -0,0 +1,42 @@
+import { asString } from '@ukef/dtfs2-common';
+import { Request } from 'express';
+import { getAndClearFieldsFromRedirectSessionData } from './get-and-clear-fields-from-redirect-session-data';
+import { getIsCheckboxChecked } from './get-is-checkbox-checked';
+import { getSelectedFeeRecordIdsFromQuery } from './get-selected-fee-record-ids-from-query';
+import { validateFacilityIdQuery } from './validate-facility-id-query';
+
+/**
+ * Extracts and processes query parameters and session data from the request.
+ *
+ * Extracts selected fee record IDs from query parameters or session data.
+ * Prioritizes query parameters if available, falling back to session data.
+ * This handles cases where the user was redirected from another page.
+ *
+ * @param req - The request object
+ * @returns An object containing extracted and processed data including
+ * facilityIdQueryString, filterError, tableDataError, and derived values
+ * such as isCheckboxChecked.
+ */
+export const extractQueryAndSessionData = (req: Request) => {
+  const { facilityIdQuery, selectedFeeRecordIds: selectedFeeRecordIdsQuery } = req.query;
+
```

Comment:
Nice extraction @sw-mattw, but could we please update the params to consume `facilityIdQuery` `selectedFeeRecordIds` params specifically, instead of passing around `req` (anti-pattern) 🙏

Reply:
Modified this in the most recent commit (ff744d7b2b317b3d125c0b15dc98fdf663078ff1) so that we don't pass in the request anymore, all of the destructuring of the request is now done in the main controller function and we only pass down the required data